### PR TITLE
DOC-145: Added documentation for APM-84

### PR DESCRIPTION
**Pull request by cpjulia**
Documentation for https://github.com/arangodb/arangodb/pull/16421.

### CODE REVIEW

File: 3.10/aql/invocation-with-arangosh.md [hunk 1]
```diff
@@ -242,6 +242,44 @@ In the ArangoDB Enterprise Edition there is an additional parameter:
...
+
+- `spillOverThresholdNumRows`: This parameter allows for input data and 
+  intermediate results to be spilled onto disk for the execution of a query 
+  after the memory usage reaches the value in bytes this parameter holds. This 
+  is used for decreasing the memory usage during the query execution. This 
+  parameter is experimental and is only taken into account if a path for a 
+  directory where the temporary data would be stored is provided with the 
+  server startup option 
+  `--temp.intermediate-results-path`. 
+  Default value: 128MB.
```
Comment (nerpaula) [inline 3.10/aql/invocation-with-arangosh.md:280]: both parameters having the same name, is that correct?

Comment (cpjulia) [reply 3.10/aql/invocation-with-arangosh.md]: no, it's `spillOverThresholdMemoryUsage`, fixed in latest commit

File: 3.10/aql/invocation-with-arangosh.md [hunk 2]
```diff
@@ -242,6 +242,44 @@ In the ArangoDB Enterprise Edition there is an additional parameter:
   into sync. The default value is 60.0 (seconds). When the max time has been reached
   the query will be stopped.
 
+Additional parameters for spilling data from the query onto disk
```
Comment (Simran-B) [inline 3.10/aql/invocation-with-arangosh.md:245]: None of the other options have their own headline. I wonder whether we should group some of the others as well, or not do that for any of them. Giving each option an own headline would be another option and allow to link to each single one (we can hide them in the ToC if necessary)

File: 3.10/aql/invocation-with-arangosh.md [hunk 3]
```diff
@@ -242,6 +242,44 @@ In the ArangoDB Enterprise Edition there is an additional parameter:
...
+-----------------------------------------------------------------
+
+  There are two additional parameters that would allow for spilling 
+  intermediate data from the query onto disk for descreasing the memory usage.
+  Note: The option of spilling data from RAM onto disk is experimental and off
```
Comment (Simran-B) [inline 3.10/aql/invocation-with-arangosh.md:250]: Use hint box

File: 3.10/release-notes-new-features310.md [hunk 4]
```diff
@@ -427,6 +427,53 @@ arangoexport \
...
 ```
 
+
+Query changes for decreasing memory usage
+-----------------------------------------
+
+With the parameters mentioned below, queries can execute with storing input
+and intermediate results temporarily on disk for decreasing of memory usage
+when it reaches a certain threshold.
+Note: this feature is experimental and off by default. The query results will
```
Comment (Simran-B) [inline 3.10/release-notes-new-features310.md:437]: Remarks should only be in the feature documentation, not the release notes. The option descriptions should probably be removed (but new start options should still need to be mentioned).

File: 3.10/aql/invocation-with-arangosh.md (hunk 2 above)
Comment (cpjulia) [reply 3.10/aql/invocation-with-arangosh.md:245]: Actually, the options above this are explained one by one starting in line 141, so I added these extra ones too

**Review by ansoboleva — APPROVED** [review]
reviewed

**Pull request merged**